### PR TITLE
Add managing Ballerina code documentation

### LINKED ISSUE #269: Update Using Modules topic
Update Using Modules page with steps to manually pull modules from Ballerina Central and use templates.

### PR DESCRIPTION
**Pull request by Asitha**
resolves #269 

## Automation tests
 - Unit tests 
   > Code coverage information
 - Integration tests
   > Details about the test cases and coverage

## Security checks
 - Followed secure coding standards in http://wso2.com/technical-reports/wso2-secure-engineering-guidelines? yes/no
 - Ran FindSecurityBugs plugin and verified report? yes/no
 - Confirmed that this PR doesn't commit any keys, passwords, tokens, usernames, or other secrets? yes/no

### CODE REVIEW

File: docs/content/src/develop/managing-ballerina-code.md [hunk 1]
```diff
@@ -0,0 +1,352 @@
+# Managing Ballerina Code
+
+When you write large programs often you would want to organize your code into shareable units. 
+In Ballerina, you can divide your program into such units which are called Ballerina modules.
```
Comment (samgnaniah) [inline docs/content/src/develop/managing-ballerina-code.md]:
```suggestion
In Ballerina, you can divide your program into such units, which are called Ballerina modules. 
```

File: docs/content/src/develop/managing-ballerina-code.md [hunk 2]
```diff
@@ -0,0 +1,352 @@
...
+$ tree .
+.
+├── Ballerina.toml
+└── src
+
+1 directory, 1 file
+
+```
+
+If we take a look at the `Ballerina.toml` file
```
Comment (samgnaniah) [inline docs/content/src/develop/managing-ballerina-code.md]:
```suggestion
Let's take a look at the `Ballerina.toml` file
```

File: docs/content/src/develop/managing-ballerina-code.md [hunk 3]
```diff
@@ -0,0 +1,352 @@
...
+```toml
+[project]
+org-name= "wso2"
+version= "0.1.0"
+
+[dependencies]
+
+```
+
+This is a manifest file. This file is used by the compiler to identify a ballerina project directory.
```
Comment (samgnaniah) [inline docs/content/src/develop/managing-ballerina-code.md]:
```suggestion
This is a manifest file. This file is used by the compiler to identify a Ballerina project directory. 
```

File: docs/content/src/develop/managing-ballerina-code.md [hunk 4]
```diff
@@ -0,0 +1,352 @@
...
+annotations, services, and objects. To share a module among programs, projects, and users you need to 
+push the module into a repository like Ballerina Central.
+
+> **Tip**: Module names can contain alphanumeric characters including dots `.`. Dots in a module name 
+> have no meaning other than the last segment after the final dot being used as a default alias within 
+> your source code.
+
+To create a Ballerina Module you can simply,
+
+- create a directory within the src directory
```
Comment (samgnaniah) [inline docs/content/src/develop/managing-ballerina-code.md]:
```suggestion
To create a Ballerina module, create a directory within the `src` directory and do one of the following: 
```

File: docs/content/src/develop/managing-ballerina-code.md [hunk 5]
```diff
@@ -0,0 +1,352 @@
...
+- use the WSO2 Ballerina Integrator VS Code plugin with predefined templates or
+- use the Ballerina CLI tool
+
+To create a Ballerina Module using the Ballerina CLI tool use `ballerina add` command. 
+
+```bash
+$ ballerina add hello-service -t service # ballerina add <module-name> -t <template-name>[:<version>]
+```
+
+> Note that you have to run this command within a ballerina project directory. Here we are creating
```
Comment (samgnaniah) [inline docs/content/src/develop/managing-ballerina-code.md]:
```suggestion
> Note that you have to run this command within a Ballerina project directory. Here we are creating 
```

File: docs/content/src/develop/managing-ballerina-code.md [hunk 6]
```diff
@@ -0,0 +1,352 @@
...
+        ├── Module.md                   # module level documentation
+        ├── resources                   # resources for the module (available at runtime)
+        └── tests                       # tests for this module (e.g. unit tests)
+            ├── hello_service_test.bal  # test file for main
+            └── resources               # resources for these tests
+
+5 directories, 4 files
+```
+
+if you open up the hello_service.bal file you can see that the module was created with a `service`
```
Comment (samgnaniah) [inline docs/content/src/develop/managing-ballerina-code.md]:
```suggestion
If you open up the `hello_service.bal` file you can see that the module was created with a `service`
```

File: docs/content/src/develop/managing-ballerina-code.md [hunk 7]
```diff
@@ -0,0 +1,352 @@
...
+        if (result is error) {
+            io:println("Error in responding", result);
+        }   
+    }   
+}
+```
+
+## Building the Project
+
+To build the project we can use the `ballerina build`:
```
Comment (samgnaniah) [inline docs/content/src/develop/managing-ballerina-code.md]:
```suggestion
To build the project we can use the `ballerina build` command:
```

File: docs/content/src/develop/managing-ballerina-code.md [hunk 8]
```diff
@@ -0,0 +1,352 @@
...
+```
+
+If you look at the code there are four phases in the build process. Namely,
+
+- compiling the source code
+- creating balos (Sharable representation of a module)
+- Running the tests
+- Generating executables
+
+If you are writing a module that does not have a `main` or a `service` entry point you wouldn't need to create the
```
Comment (samgnaniah) [inline docs/content/src/develop/managing-ballerina-code.md]:
```suggestion
If you are writing a module that does not have a `main` or a `service` entry point, you wouldn't need to create the
```

File: docs/content/src/develop/managing-ballerina-code.md [hunk 9]
```diff
@@ -0,0 +1,352 @@
...
+In that case, you can instruct the Ballerina CLI tool to ignore the executable creation phase by 
+providing the `-c` argument.
+
+```bash
+$ ballerina build hello_service -c
+``` 
+
+## Ballerina Central: Public Module Repository
+
+Once you have build your Ballerina project you can store your modules in a public directory called
```
Comment (samgnaniah) [inline docs/content/src/develop/managing-ballerina-code.md]:
```suggestion
Once you have built your Ballerina project, you can store your modules in a public directory called 
```

File: docs/content/src/develop/managing-ballerina-code.md [hunk 10]
```diff
@@ -0,0 +1,352 @@
...
+>
+> ```toml
+> [proxy]
+> host = "localhost"
+> port = "3128"
+> username = ""
+> password = ""
+> ```
+
+Once Settings.toml and the Ballerina.toml is updated you are ready to push your module into Ballerina Central.
```
Comment (samgnaniah) [inline docs/content/src/develop/managing-ballerina-code.md]:
```suggestion
Once `Settings.toml` and `Ballerina.toml` are updated, you are ready to push your module into Ballerina Central.
```

File: docs/content/src/develop/managing-ballerina-code.md [hunk 11]
```diff
@@ -0,0 +1,352 @@
...
+  path = "../utils/target/classes/lib/javax.jms-api-2.0.1.jar"
+  artafactId = "javax.jms-api"
+  version = "2.0.1"
+  groupId = "javax.jms"
+```
+
+## Further Reading
+
+You can read more about structuring your Ballerina code on Ballerina Documentation, 
+[How to Structure Ballerina Code](https://v1-0.ballerina.io/learn/how-to-structure-ballerina-code/)
```
Comment (samgnaniah) [inline docs/content/src/develop/managing-ballerina-code.md]:
```suggestion
You can read more about structuring your Ballerina code in the [How to Structure Ballerina Code](https://v1-0.ballerina.io/learn/how-to-structure-ballerina-code/) topic in the Ballerina documentation.
```

File: docs/content/src/develop/managing-ballerina-code.md [hunk 12]
```diff
@@ -0,0 +1,352 @@
...
+"wso2/gsheets4" = { path = "path/to/gsheets4.balo", version = "0.9.1"}
+```
+
+> Once you compile the dependent project, the relevant balo file is created within the `target/balo` directory
+> of that project. You can give either the relative path or the absolute path to the balo file of the dependent 
+> project from your project.
+
+### Platform Dependencies
+
+With jBallerina you can depend on java libraries and interoperate with them. To do that and to make the
```
Comment (samgnaniah) [inline docs/content/src/develop/managing-ballerina-code.md]:
```suggestion
With jBallerina you can depend on Java libraries and interoperate with them. To do that and to make the 
```

File: docs/content/src/develop/managing-ballerina-code.md [hunk 13]
```diff
@@ -0,0 +1,352 @@
...
+```
+
+> Once you compile the dependent project, the relevant balo file is created within the `target/balo` directory
+> of that project. You can give either the relative path or the absolute path to the balo file of the dependent 
+> project from your project.
+
+### Platform Dependencies
+
+With jBallerina you can depend on java libraries and interoperate with them. To do that and to make the 
+compiler aware of the java libraries we have to update the `Ballerina.toml` file.
```
Comment (samgnaniah) [inline docs/content/src/develop/managing-ballerina-code.md]:
```suggestion
compiler aware of the Java libraries we have to update the `Ballerina.toml` file.
```

File: docs/content/src/develop/managing-ballerina-code.md [hunk 14]
```diff
@@ -0,0 +1,352 @@
...
+import wso2/http;
+
+public function main() {
+  http:Person x = http:getPerson();
+}
+```
+
+### Path Dependencies
+
+When you want to depend on a module that you haven't uploaded to a remote repository you can use
```
Comment (samgnaniah) [inline docs/content/src/develop/managing-ballerina-code.md]:
```suggestion
When you want to depend on a module that you have not uploaded to a remote repository, you can use 
```

File: docs/content/src/develop/managing-ballerina-code.md [hunk 15]
```diff
@@ -0,0 +1,352 @@
...
+
+When you import a module, you can use its functions, annotations, and other objects in your code. You can also 
+reference the objects with a qualified identifier, followed by a colon `:`. 
+
+For example, `<identifier>:<module-object>`.
+
+Identifiers are either derived or explicit. The default identifier is either the module name, or if the module name 
+has dots `.` included, then the last word after the last dot. 
+
+For example, `import ballerina/http;` will have `http:` be the derived identifier.
```
Comment (samgnaniah) [inline docs/content/src/develop/managing-ballerina-code.md]:
```suggestion
For example, `import ballerina/http;` will have `http:` as the derived identifier. 
```

File: docs/content/src/develop/managing-ballerina-code.md (hunk 4 above)
Comment (Asitha) [reply docs/content/src/develop/managing-ballerina-code.md]: `- create a directory within the src directory ` is an option too. So we can't merge the lines, Shall I put an or after that line?

Comment (samgnaniah) [reply docs/content/src/develop/managing-ballerina-code.md]: Maybe better to say something like "To create a Ballerina module, do one of the following:". After that we can introduce the list without using "or" at all.

File: docs/content/src/develop/managing-ballerina-code.md [hunk 16]
```diff
@@ -0,0 +1,352 @@
...
+  module = "jms"
+  path = "../utils/target/classes/lib/javax.jms-api-2.0.1.jar"
+  artafactId = "javax.jms-api"
+  version = "2.0.1"
+  groupId = "javax.jms"
+```
+
+## Further Reading
+
+You can read more about structuring your Ballerina code on Ballerina Documentation,
```
Comment (samgnaniah) [inline docs/content/src/develop/managing-ballerina-code.md]: Can remove this line.

**Pull request merged**